src/traces/image/defaults.js

```diff
@@ -0,0 +1,36 @@
+/**
+* Copyright 2012-2019, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+
+'use strict';
+
+var Lib = require('../../lib');
+var attributes = require('./attributes');
+var constants = require('./constants');
+
+module.exports = function supplyDefaults(traceIn, traceOut) {
+    function coerce(attr, dflt) {
+        return Lib.coerce(traceIn, traceOut, attributes, attr, dflt);
+    }
+    var z = coerce('z');
+    if(z === undefined || !z.length) {
```

Comment:
We might want to check that `z[0]` and `z[0].length` are also truthy here. I can't imagine having 1-dimensional raster data - am I right?

Reply:
That could be useful option with typed arrays.

Reply:
Done in 73c0740f53fc82a70196d1993ba2824db2be85a3

Reply:
Thanks!